src/AutoMapper/Mappers/ExpressionMapper.cs

```diff
@@ -247,7 +247,13 @@ private Type GetSourceType(PropertyMap propertyMap)
             {
                 if (propertyMap.SourceMember == null)
                 {
-                    return propertyMap.SourceType;
+                    return propertyMap.SourceType ??
```

Comment:
I would think `var sourceType = propertyMap.SourceType;` would be sufficient.  If not similar to what @BlaiseD has with for source type but include CustomResolver as it's a LambdaExpression as well.

`PropertyMap.SourceType` has conditions for all types of resolvers types, so that should be used.  If it's null it will return null, but it shouldn't happen unlike what @lbargaoanu says.

The problem is it's a custom expression and `SourceMember.GetMemberType()` doesn't account for custom expressions.  I would think CustomResolver could work as well because it's a lambda but might have to be translated.

Might suggest what @BlaiseD says for setting sourceType for now and can latter test Custom Resolvers and ValueResolverConfig to see if they are supported in UseAsDataSource.